nautobot/docs/plugins/development.md

```diff
@@ -1146,3 +1146,147 @@ try:
+
+`NautobotUIViewSet`'s `lookup_field` is default to `slug` as we transition from `pk` to `slug` across Nautobot Apps. If you want to use a different field to look up an object, just override the default `lookup_field` in your ViewSet attributes:
+
+```python
+from nautobot.core.views.viewsets import NautobotUIViewset
+
+class YourPluginModelUIViewSet(NautobotUIViewSet):
+    ...
+    lookup_field = "pk"
+    ...
+```
+
+### Excluding ViewMixins from NautobotUIViewSet
+
+For plugin models that do not require certain views, simply inherit directly from the `ViewMixins` available in `nautobot.core.views.mixins` instead of `NautobotUIViewSet`.
+
+Concrete examples for excluding `ViewMixins`, checkout `CircuitTerminationUIViewSet` and `CircuitTypeUIViewSet` in `nautobot.circuits.views`.
+
+```python
+## A plugin model viewset that does not support bulk views and operations
+from nautobot.core.views import mixins as view_mixins
+
+class YourPluginModelUIViewSet(
+    view_mixins.ObjectListViewMixin,
+    view_mixins.ObjectDetailViewMixin,
+    view_mixins.ObjectEditViewMixin,
+    view_mixins.ObjectDestroyViewMixin,
+):
+
+    filterset_class = YourPluginModelFilterSet
+    filterset_form_class = YourPluginModelFilterForm
+    form_class = YourPluginModelForm
+    queryset = YourPluginModel.objects.all()
+    serializer_class = serializers.YourPluginModelSerializer
+    table_class = YourPluginModelTable
+    # You do not need to specify attributes that are not needed.
+```
+
+### Template Naming for NautobotUIViewSet
+
+Template naming is very intuitive in NautobotUIViewSet. In `templates/yourpluginmodel` folder, name your templates following this convention `{app_label}/{model_name}_{self.action}.html`.
+
+| ViewMixins    | self.action   |
+| ------------- |:-------------:|
+| ObjectlistViewMixin        | list         |
+| ObjectDetailViewMixin      | retrieve     |
+| ObjectEditViewMixin        | create/update|
+| ObjectDestroyViewMixin     | destroy      |
+| ObjectBulkDestroyViewMixin | bulk_destroy |
+| ObjectBulkCreateViewMixin  | bulk_create  |
+| ObjectBulkUpdateViewMixin  | bulk_update  |
+
+For example, for a DetailView template for `yourpluginmodel`, the template name will be `yourplugin/yourpluginmodel_retrieve.html`, for a BulkCreateView template for `yourpluginmodel`, the template name will be `yourplugin/yourpluginmodel_bulk_create.html` and etc.
+
+If you do not specify your own templates in `templates/yourpluginmodel` folder, `NautobotUIViewSet` will fall back to `generic/object_{self.action}.html`.
+
+If you do specify a `{app_label}/{model_opts.model_name}_create.html` file but not a `{app_label}/{model_opts.model_name}_update.html` file. When you update an object, it will fall back to `{app_label}/{model_opts.model_name}_create.html` and vice versa.
```

Comment:
```suggestion
Since in many cases the `create` and `update` templates for a model will be identical, you are not required to create both. If you provide a `{app_label}/{model_opts.model_name}_create.html` file but not a `{app_label}/{model_opts.model_name}_update.html` file, then when you update an object, it will fall back to `{app_label}/{model_opts.model_name}_create.html` and vice versa.
```